pkg/crc/machine/start.go

```diff
@@ -636,3 +645,37 @@ func logBundleDate(crcBundleMetadata *bundle.CrcBundleInfo) {
 		}
 	}
 }
+
+func ensureRoutesControllerIsRunning(sshRunner *crcssh.Runner, ocConfig oc.Config) error {
+	bin, err := json.Marshal(v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "routes-controller",
+			Namespace: "openshift-ingress",
+		},
+		Spec: v1.PodSpec{
+			ServiceAccountName: "router",
+			Containers: []v1.Container{
+				{
+					Name:            "routes-controller",
+					Image:           "quay.io/crcont/routes-controller:6effc2f0304d3a11c1e22c278b10630e834d0220",
+					ImagePullPolicy: v1.PullIfNotPresent,
+				},
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if err := sshRunner.CopyData(bin, "/tmp/routes-controller.json", 0444); err != nil {
```

Comment:
@guillaumerose this is only pod definition so if somehow this pod crash then this route stuff will break, should we have deployment as `Kind` instead pod so that pod recreation happen if pod crashes?

Reply:
Deployment are useful version management. For crash, there is the RestartPolicy on the pod directly. It is "Always" by default so we should be good. 

I tried to kill the process by hand. It is restarted. 
I tried to start/stop/start the VM. The process is here.

Reply:
@guillaumerose For me it is not happening.
```
13:07 $ oc get pods -n openshift-ingress
NAME                              READY   STATUS    RESTARTS   AGE
router-default-7476bbd498-g9r7k   1/1     Running   1          13d
routes-controller                 1/1     Running   0          30m

13:08 $ oc delete pod routes-controller -n openshift-ingress
pod "routes-controller" deleted

13:11 $ oc get pods -n openshift-ingress
NAME                              READY   STATUS    RESTARTS   AGE
router-default-7476bbd498-g9r7k   1/1     Running   1          13d
```

Reply:
Yes sure, I mean kill with `sudo kill -9 pid`. 
If the user does this, why bother him and recreate the pod ?

Reply:
Since it is part of `openshift-ingress` and if  a user just delete all the pods under this namespace, all pods except this one come up without any issue, Also can we make it as static pod if we really want it to be running always?

Reply:
I tried to add in static pod directory but it doesn't work:

```
Apr 19 12:19:17 crc-xl2km-master-0 hyperkube[2154]: E0419 12:19:17.473167    2154 kubelet.go:1671] Failed creating a mirror pod for "routes-controller-crc-xl2km-master-0_openshift-ingress(fd89789fbe33ba7075234cb104ec8e39)": pods "routes-controller-crc-xl2km-master-0" is forbidden: a mirror pod may not reference service accounts
```

>  if a user just delete all the pods under this namespace

It should not happen or he really wants to do this?

Reply:
This is not blocker for this PR so let's have some follow up issue in case we hit it.